src/dune/action_exec.mli

```diff
@@ -1,15 +1,21 @@
 open! Stdune
 
-module Context : sig
-  type t
+type done_or_more_deps =
+  | Done
+  (* TODO jstaron: There is a problem with having a Set/Map of
```

Comment:
Yeah, this is indeed weird. I guess we could make the keys be of the form `cwd * Dependency.t`, and then only pass the dep to the action if the cwd matches. It's probably not worth complicating this too much though, let's just forbid multiple `dynamic-run` within a single rule for now and leave a comment here as a warning for future refactorers.

Reply:
Done